src/EnergyPlus/RuntimeLanguageProcessor.cc

```diff
@@ -1771,743 +1768,904 @@ ErlValueType EvaluateExpression(EnergyPlusData &state, int const ExpressionNum,
         if (ReturnValue.Type != Value::Error) {
 
             // Perform the operation
-            {
-                auto const SELECT_CASE_var(state.dataRuntimeLang->ErlExpression(ExpressionNum).Operator);
-
-                if (SELECT_CASE_var == ErlFunc::Literal) {
-                    ReturnValue = Operand(1);
-                    ReturnValue.initialized = true;
-                } else if (SELECT_CASE_var == ErlFunc::Negative) { // unary minus sign.  parsing does not work yet
-                    ReturnValue = SetErlValueNumber(-1.0 * Operand(1).Number);
-                } else if (SELECT_CASE_var == ErlFunc::Divide) {
-                    if ((Operand(1).Type == Value::Number) && (Operand(2).Type == Value::Number)) {
-                        if (Operand(2).Number == 0.0) {
-                            ReturnValue.Type = Value::Error;
-                            ReturnValue.Error = "EvaluateExpression: Divide By Zero in EMS Program!";
-                            if (!state.dataGlobal->DoingSizing && !state.dataGlobal->KickOffSimulation &&
-                                !state.dataEMSMgr->FinishProcessingUserInput) {
-                                seriousErrorFound = true;
-                            }
-                        } else {
-                            ReturnValue = SetErlValueNumber(Operand(1).Number / Operand(2).Number);
+
+            switch (state.dataRuntimeLang->ErlExpression(ExpressionNum).Operator) {
```

Comment:
On second thought, maybe put this back as an `if-else` ladder. Maybe keeping the ERL path slow will encourage people to switch to the Python path. :grin: (by the way, there was an interesting presentation by IBACOS a few weeks ago where they prototyped this new modular home ducting system with a multi-zone "manifold" type of hub and were able to model its performance using Python EMS, it was pretty sweet! I should try to get the presentation)